teps/0048-task-results-without-results.md

```diff
@@ -0,0 +1,425 @@
+and can design it such that the `params` are mapped to `pipeline` params or initialized with the `results` of other
+tasks in a `pipeline`. At the Task level, parameters can be declared as optional. The Task author may provide a default
+value for optional parameters. The Pipeline author must populate every parameter that is not declared as optional in the Task.
+
+This proposal introduces a new field called `results` at the `pipelineTask` level. `results` can have a list of result
+variables, and their default values.
+
+In the following example, `pipelineTask` `add` has specified `results` with the default value. Now, `x` is referring
+to a task result of `add`. If `add` fails for some reason, finally task `multiply` continues execution with default
+`sum` which is `0`:
+
+```yaml
+kind: Pipeline
+spec:
+  tasks:
+    - name: add
+      taskRef:
+        name: add-task
+      params:
+        - name: first
+          value: $(params.first)
+        - name: second
+          value: $(params.second)
+      results:
+        - name: sum
+          default: 0
+  finally:
+    - name: multiply
+      taskRef:
+        name: multiply
+      params:
+        - name: x
+          value: $(task.add.results.sum)
+```
+
+### User Stories
+
+#### Story 1
+
+For a cleanup use case, `finally` task receives the project name, or the configuration name as a `task result` from
+the `acquire` task. If `acquire` task fails to acquire any resources, `cleanup` is skipped.
+
+#### Story 2
+
+Pipeline that builds a container image and deploys it. The pipeline accepts an image reference parameter.
+
+* The first task looks up in the registry if the image exists.
+
+* The second task builds the container images. It is guarded via `when` and only runs if the first task reports that the
+  image is missing.
+
+* The third task takes the image reference from a result of the second task and deploys the container.
+
+The pipeline authors sets a default value for the image reference of it's own image reference input param, so that
+if task2 is `skipped`, task3 can deploy the image that was passed to the pipeline by the `pipelineRun`.
+
+```yaml
+kind: Pipeline
+spec:
+  params:
+    - name: targetImage
+      description: the image to deploy (and build if missing)
+  tasks:
+    - name: lookup-image
+      taskRef:
+        name: search-image
+      params:
+        - name: imageRef
+          value: "$(params.targetImage)"
+    - name: build-image
+      when:
+        - input: "$(tasks.lookup-image.results.found)"
+          operator: in
+          values: ["False"]
+      taskRef:
+        name: build-image
+      params:
+        - name: imageRef
+          value: "$(params.targetImage)"
+      results:
+        - name: builtImage
+          default: "$(params.targetImage)"
+    - name: deploy-image
+      taskRef:
+        name: deploy-image
+      params:
+        - name: imageRef
+          value: "$(tasks.build-image.results.builtImage)"
+```
+
+
+## Design Details
```

Comment:
can we include pros and cons for the proposed option? (seems like we have them for the alternatives)